airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -573,6 +573,15 @@ def build_pod_request_obj(self, context=None):
         )
         return pod
 
+    def dry_run(self) -> None:
+        """
+        Prints out the pod definition that would be created by this operator.
+        Does not include labels specific to the task instance (since there isn't
+        one in a dry_run) and excludes all empty elements.
+        """
+        pod = self.build_pod_request_obj()
```

Comment:
Do you think we should use `pod_mutation_hook` on the POD object:

https://github.com/apache/airflow/blob/e07e8319465ea4598791b6b61b5fe7c46f159f86/airflow/settings.py#L184-L196

https://github.com/apache/airflow/blob/e07e8319465ea4598791b6b61b5fe7c46f159f86/airflow/providers/cncf/kubernetes/utils/pod_launcher.py#L102-L104

or show an original one, and the one with the changes from pod_mutation_hook 🤷  -- I have no strong opinion on this but just wanted to open it up for discussion

Reply:
@BasPH @jedcunningham wdyt?

Reply:
oh with regard to dry run... i think you're right we probably should run it through the mutation hook.  

you know... this is interesting.... mutating the pod in launcher is  actually not great i don't think.  we should mutate it in `build_pod_request_obj` 

ideally, the pod returned from `build_pod_request_obj` should be the actual pod requested.

wdyt @kaxil

Reply:
Yeah, I agree, atleast based on the current code. It's only used in the following places:

```
airflow/cli/commands/kubernetes_command.py:30:from airflow.settings import pod_mutation_hook
airflow/cli/commands/kubernetes_command.py:59:        pod_mutation_hook(pod)
airflow/executors/kubernetes_executor.py:46:from airflow.settings import pod_mutation_hook
airflow/executors/kubernetes_executor.py:252:        pod_mutation_hook(pod)
airflow/kubernetes/pod_launcher_deprecated.py:36:from airflow.settings import pod_mutation_hook
airflow/kubernetes/pod_launcher_deprecated.py:92:        pod_mutation_hook(pod)
airflow/models/taskinstance.py:2024:        settings.pod_mutation_hook(pod)
airflow/providers/cncf/kubernetes/utils/pod_launcher.py:39:from airflow.settings import pod_mutation_hook
airflow/providers/cncf/kubernetes/utils/pod_launcher.py:104:        pod_mutation_hook(pod)
```

Reply:
OK -- For KPO, I will move the mutation hook from launcher to KPO `build_pod_request_obj`

Then we don't need to run the hook as part of dry run

So we don't need to change this PR. 

Sound good?

For now I'll resolve this convo, and i'll post the other PR soon